components/autocomplete/SelectedItems.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const SelectedItems = ({ selected, onRemove }) => {
+    const handleRemove = (removed) => onRemove(removed, selected.filter((item) => item !== removed));
```

Comment:
His parent should manage the `selected` list. `onRemove` should just forward the `index` to his parent. The idea is to not have a state in `SelectedItems`, his role is just to display the list.